### PR TITLE
[DUOS-2277][risk=no] New API to replace existing one for user relevant datasets

### PR DESCRIPTION
**Pull request by rushtong**
## Addresses
Partially addresses https://broadworkbench.atlassian.net/browse/DUOS-2277

Related to [DUOS-1929](https://broadworkbench.atlassian.net/browse/DUOS-1929)
This covers API work that was discovered to be missing for chairs and members while working on a similar page for DAC-dataset-approval. This PR creates a new version of the API to get relevant datasets for a user but returning full `Dataset` objects instead of the deprecated `DatasetDTO` objects. The one main difference between the two are how they treat the `alias` value. `Dataset` treats it as a numeric value and adds a `datasetIdentifier` field for the parsed version of the numeric `alias` value while the DTO puts the identifier into the alias.

See also https://github.com/DataBiosphere/duos-ui/pull/1999 where there is some UI work related to this.

----
Have you read [CONTRIBUTING.md](../CONTRIBUTING.md) lately? If not, do that first.

- Label PR with a Jira ticket number and include a link to the ticket
- Label PR with a security risk modifier [no, low, medium, high]
- PR describes scope of changes
- Get a minimum of one thumbs worth of review, preferably two if enough team members are available
- Get PO sign-off for all non-trivial UI or workflow changes
- Verify all tests go green
- Test this change deployed correctly and works on dev environment after deployment

[DUOS-1929]: https://broadworkbench.atlassian.net/browse/DUOS-1929?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/main/java/org/broadinstitute/consent/http/db/DatasetDAO.java [hunk 1]
```diff
@@ -456,6 +455,18 @@ void updateDataset(@Bind("datasetId") Integer datasetId,
...
+    @SqlQuery("""
+            SELECT distinct d.*, k.key, p.property_value, c.consent_id, d.dac_id, c.translated_use_restriction
+            FROM dataset d
+            LEFT JOIN dataset_property p ON p.dataset_id = d.dataset_id
+            LEFT JOIN dictionary k ON k.key_id = p.property_key
+            LEFT JOIN consent_associations a ON a.dataset_id = d.dataset_id
+            LEFT JOIN consents c ON c.consent_id = a.consent_id
+            WHERE d.dac_id IN (<dacIds>)
+        """)
+    List<Dataset> findDatasetListByDacIds(@BindList("dacIds") List<Integer> dacIds);
```
Comment (nmalfroy) [inline src/main/java/org/broadinstitute/consent/http/db/DatasetDAO.java:468]: I just want to say I really like this dao pattern

File: src/main/java/org/broadinstitute/consent/http/resources/UserResource.java [hunk 2]
```diff
@@ -136,6 +137,24 @@ public Response getDatasetsFromUserDacs(@Auth AuthUser authUser) {
...
+    @Produces("application/json")
+    @RolesAllowed({CHAIRPERSON, MEMBER})
+    public Response getDatasetsFromUserDacsV2(@Auth AuthUser authUser) {
+        try {
+            User user = userService.findUserByEmail(authUser.getEmail());
+            List<Integer> dacIds = user.getRoles().stream()
+                .map(UserRole::getDacId)
+                .filter(Objects::nonNull)
+                .collect(Collectors.toList());
+            List<Dataset> datasets = dacIds.isEmpty() ? List.of() : datasetService.findDatasetListByDacIds(dacIds);
```
Comment (nmalfroy) [inline src/main/java/org/broadinstitute/consent/http/resources/UserResource.java:151]: Would it make sense to throw a 404 if the list is empty?

**Review by nmalfroy — APPROVED** [review]
Looks good!  Just that one question about returning an specific error

File: src/main/java/org/broadinstitute/consent/http/db/DatasetDAO.java (hunk 1 above)
Comment (rushtong) [reply src/main/java/org/broadinstitute/consent/http/db/DatasetDAO.java:468]: I'm enjoying this too - it's so much easier to read as well as test if you have a db app up and running.

File: src/main/java/org/broadinstitute/consent/http/resources/UserResource.java (hunk 2 above)
Comment (rushtong) [reply src/main/java/org/broadinstitute/consent/http/resources/UserResource.java:151]: Great point, I think that does make sense.

**Review by fboulnois — APPROVED** [review]
looks good!

**Pull request merged**